eventstore/memory/memory.go

```diff
@@ -124,10 +108,21 @@ func (e *Memory) GlobalEvents(start, count uint64) ([]core.Event, error) {
 	return events, nil
 }
 
-// Close does nothing
-func (e *Memory) Close() {}
+func (m *Memory) All(start core.Version, count uint64) func() (core.Iterator, error) {
+	return func() (core.Iterator, error) {
+		// Get 10 at a time
+		events, err := m.globalEvents(start, count)
+		if err != nil {
+			return nil, err
+		}
 
-// aggregateKey generate a aggregate key to store events against from aggregateType and aggregateID
-func aggregateKey(aggregateType, aggregateID string) string {
-	return aggregateType + "_" + aggregateID
+		// no events to fetch
+		if len(events) == 0 {
+			return core.ZeroIterator{}, nil
+		}
+
+		// next time the function is called it start from the last fetched event +1
```

Comment:
Is "it" here a reference to "iterator", or is it is missing a "will"?
```suggestion
		// next time the function is called it will start from the last fetched event +1
```